server/links/datatrails/readme.md

```diff
@@ -0,0 +1,253 @@
+
+To align with SCITT semantics, the vcon_uuid is set to the DataTrails `subject` event attribute. (`event_attributes.subject`)
+
+To query the history of DataTrails Events for a given vCon, use the following:
+
+- For bash/curl commands, configure the `.datatrails/bearer-token.txt` using the DataTrails [Creating Access Tokens](https://docs.datatrails.ai/developers/developer-patterns/getting-access-tokens-using-app-registrations/) developer docs.
+- Query the collection of DataTrails Events, using the `subject` attribute.
+  Set the `VCON` env variable to the `vcon_uuid`
+
+   ```bash
+   DATATRAILS_EVENTS_URL="https://app.datatrails.ai/archivist/v2/assets/-/events"
+   VCON=bbba043b-d1aa-4691-8739-ac3ddd030390
+   curl -g -X GET \
+     -H "@$HOME/.datatrails/bearer-token.txt" \
+     "$DATATRAILS_EVENTS_URL?event_attributes.subject=vcon://$VCON" \
+     | jq
+   ```
+
+- Verify Inclusions of a Specific vCon Hash
+
+   ```bash
+   DATATRAILS_EVENTS_URL="https://app.datatrails.ai/archivist/v2/assets/-/events"
+   VCON="bbba043b-d1aa-4691-8739-ac3ddd030390"
+   VCON_HASH="eae12ce2ae12c7b1280921236857d2dc1332babd311ae0fbcab620bdb148fd0d"
+   curl -g -X GET \
+     -H "@$HOME/.datatrails/bearer-token.txt" \
+     "$DATATRAILS_EVENTS_URL?event_attributes.subject=vcon://$VCON" \
+     + "&event_attributes.payload_hash_alg=SHA-256" \
+     + "&event_attributes.payload_hash_value=$VCON_HASH" \
+     | jq
+   ```
+
+- Query Events for a Specific vCon for a Specific Operation
+
+   ```bash
+   DATATRAILS_EVENTS_URL="https://app.datatrails.ai/archivist/v2/assets/-/events"
+   VCON="bbba043b-d1aa-4691-8739-ac3ddd030390"
+   VCON_OPERATION="vcon-create"
+   curl -g -X GET \
+     -H "@$HOME/.datatrails/bearer-token.txt" \
+     "$DATATRAILS_EVENTS_URL?event_attributes.subject=vcon://$VCON" \
+     + "&event_attributes.vcon_operation=$VCON_OPERATION" \
+     | jq
+   ```
+
+- Query All Events for a Specific Operations
+
+   ```bash
+   DATATRAILS_EVENTS_URL="https://app.datatrails.ai/archivist/v2/assets/-/events"
+   VCON_OPERATION="vcon-create"
+   curl -g -X GET \
+     -H "@$HOME/.datatrails/bearer-token.txt" \
+     "$DATATRAILS_EVENTS_URL?event_attributes.vcon_operation=$VCON_OPERATION" \
+     | jq
+   ```
+
+- Limit Events Created by a Specific DataTrails Identity
+
+   ```bash
+   DATATRAILS_EVENTS_URL="https://app.datatrails.ai/archivist/v2/assets/-/events"
+   VCON=bbba043b-d1aa-4691-8739-ac3ddd030390
+   curl -g -X GET \
+     -H "@$HOME/.datatrails/bearer-token.txt" \
+     "$DATATRAILS_EVENTS_URL?event_attributes.subject=vcon://$VCON" \
+     + "&principal_declared.issuer=https://app.datatrails.ai/appidpv1" \
+     + "&principal_declared.subject=b5cfacfd-b918-4338-ad61-f4947477f874" \
+     | jq
+   ```
+
+## Development and Testing
+
+### Running Tests
+
+To run the tests for the DataTrails Asset Link:
+
+1. Install pytest:
+
+   ```bash
+   pip install pytest
+   ```
+
+2. Run the tests:
+
+   ```bash
+   pytest datatrails-link-tests.py
```

Comment:
@SteveLasker this seems to be outdated. You should be able to just run `pytest` and it should find tests in the current folder and it subfolders.

Also, can you please name that test `test_datatrails_link.py` or something like that, so it will explicitly contains name of the link and it will be easy to see which tests failed when we run entire test suite for the project.

Reply:
Ahh, yeah, I can rename them back. I misunderstood your previous comment to name everything `test_` to match the functions or even file names.